axelor-account/src/main/java/com/axelor/apps/account/db/repo/InvoiceManagementRepository.java

```diff
@@ -50,6 +55,8 @@ public Invoice copy(Invoice entity, boolean deep) {
   @Override
   public Invoice save(Invoice invoice) {
     try {
+      invoiceValidationService.checkNotOnlyNonDeductibleTaxes(invoice);
+      invoiceValidationService.checkSumOfNonDeductibleTaxes(invoice);
```

Comment:
I think it would be better to do these checks inside the method computeInvoice from InvoiceGenerator